python/tskit/tables.py

```diff
@@ -2258,6 +2258,32 @@ def __eq__(self, other):
     def __getstate__(self):
         return self.asdict()
 
+    @classmethod
+    def load(cls, file_or_path):
+        file, local_file = util.convert_file_like_to_open_file(file_or_path, "rb")
+        ll_tc = _tskit.TableCollection(1)
+        ll_tc.load(file)
+        tc = TableCollection(1)
+        tc._ll_tables = ll_tc
+        return tc
+
+    def dump(self, file_or_path, skip_checks=False):
```

Comment:
I'm not sure we want the ``skip_checks`` argument here as it might have some unwanted effects, and might be expensive. I think we should just write out the state as it is.

If we do have the checks we should probably implement at the C level using the check integrity function, rather than just calling ``self.tree_sequence()``.

Reply:
Removed in https://github.com/tskit-dev/tskit/pull/986/commits/82edb1fa6a1c944a86ce731d14fecf13df0103d3